pkg/ovssubnet/common.go

```diff
@@ -597,9 +604,25 @@ func (oc *OvsController) watchServices(ready chan<- bool, start <-chan string) {
 			}
 			switch ev.Type {
 			case api.Added:
-				oc.flowController.AddServiceOFRules(netid, ev.Service.IP, ev.Service.Protocol, ev.Service.Port)
+				oc.services[ev.Service.UID] = ev.Service
+				for _, port := range ev.Service.Ports {
+					oc.flowController.AddServiceOFRules(netid, ev.Service.IP, port.Protocol, port.Port)
+				}
 			case api.Deleted:
-				oc.flowController.DelServiceOFRules(netid, ev.Service.IP, ev.Service.Protocol, ev.Service.Port)
+				delete(oc.services, ev.Service.UID)
+				for _, port := range ev.Service.Ports {
+					oc.flowController.DelServiceOFRules(netid, ev.Service.IP, port.Protocol, port.Port)
+				}
+			case api.Modified:
```

Comment:
We can do additional check here. If old service IP/protocol/ports match with new service then ignore updating OF rules. This means service update is not relevant to us, update related to SessionAffinity/ExternalIPs/Selector or some other fields.